pkg/storegateway/series_refs.go

```diff
@@ -641,6 +700,9 @@ func (s *loadingSeriesChunkRefsSetIterator) Next() bool {
 		return s.Next() // we didn't find any suitable series in this set, try with the next one
 	}
 	s.currentSet = nextSet
+	if s.skipChunks {
+		storeCachedSeries(s.ctx, s.indexCache, s.tenantID, s.blockID, s.matchers, s.shard, seriesCacheKey, extractLabelsFromSeriesChunkRefs(nextSet.series), s.logger)
```

Comment:
Assuming you follow my suggestion and keep the non-streaming series cache implementation as it is in `main` (so we don't touch it) and we add a new functions on the cache interface (.e.g. `(Fetch|Store)SeriesFromPostings()`), couldn't we just marshal `seriesChunkRefsSet` here instead of having to do `extractLabelsFromSeriesChunkRefs()`?